src/Components/SmartComponents/Reports/DownloadCVEsReport.js

```diff
@@ -35,31 +44,45 @@ const DownloadCVEsReport = ({ filters, params, reportData, buttonProps, isReport
             }
         }));
 
-        const firstPage = firstPagePDF({ cves: data.slice(0, 13), meta, filters, intl, isReportDynamic, reportData, user });
+        const firstPage = firstPagePDF({ data: data.slice(0, 13), meta, filters, intl, isReportDynamic, reportData, user, type });
 
-        const otherPages = data.slice(13, data.length).reduce((resultArray, item, index) => {
-            const chunkIndex = Math.floor(index / 23);
-            !resultArray[chunkIndex] && (resultArray[chunkIndex] = []);
-            resultArray[chunkIndex].push(item);
+        const otherPages = otherPagesPDF(data);
 
-            return resultArray;
-        }, []);
+        setLoading(false);
+
+        return [firstPage, ...otherPages.map((cvePage, index) =>
+            tablePage({ data: cvePage, page: index, intl, header: reportData && reportData.header, type }))];
+    };
+
+    const systemsFetch = async () => {
+        setLoading(true);
+
+        let { data, meta } = await getSystems({
+            ...params,
+            ...DownloadReport.defaultParams
+        });
+
+        const firstPage = firstPagePDF({ data: data.slice(0, 13), meta, filters, intl });
+
+        const otherPages = otherPagesPDF(data);
 
         setLoading(false);
 
-        return [firstPage, ...otherPages.map((pageCves, index) =>
-            tablePage({ data: pageCves, page: index, intl, header: reportData && reportData.header }))];
+        return [firstPage, ...otherPages.map((systemPage, index) =>
+            tablePage({ data: systemPage, page: index, intl, header: reportData && reportData.header }))];
     };
 
     return (
         <div>
             <DownloadButton
                 label={loading ? intl.formatMessage(messages.loading) : intl.formatMessage(messages.kebabExportAsPDF)}
-                asyncFunction={dataFetch}
+                asyncFunction={type === 'cves' ? cvesFetch : systemsFetch}
```

Comment:
This component it's about to be complex and it loses the single responsibility principle. (download cves AND systems)   For now, we have only 2 "types" cves and system but it can easily be more complex. 

it would be better to create an abstract component to handle the download